scripts/lint/metadata.ts

```diff
@@ -20,35 +20,81 @@ export const verifyMetadata = async (
   const { metadata, errors: parsingErrors } = usercssMeta.parse(content, {
     allowErrors: true,
   });
+  const lines = content.split("\n");
 
   // Pretty print / annotate the parsing errors.
   parsingErrors.map((e) => {
     let startLine = 0;
-    for (const line of content.split("\n")) {
+    for (const line of lines) {
       startLine++;
       e.index -= line.length + 1;
       if (e.index < 0) break;
     }
     log(e.message, { file, startLine, content });
   });
 
-  for (const [key, value] of Object.entries(assert)) {
-    const defacto = metadata[key];
-    if (defacto !== value) {
-      const line = content
-        .split("\n")
+  for (const [key, expected] of Object.entries(assert)) {
+    const current = metadata[key];
+
+    if (current !== expected) {
+      const line = lines
         .findIndex((line) => line.includes(key)) + 1;
 
+      const message = current === undefined
+        ? sprintf("Metadata `%s` should not be undefined", color.bold(key))
+        : sprintf(
+          'Metadata `%s` should be "%s" but is "%s"',
+          color.bold(key),
+          color.green(expected),
+          color.red(String(current)),
+        );
+
+      log(message, {
+        file,
+        startLine: line !== 0 ? line : undefined,
+        content,
+      }, "warning");
+    }
+  }
+
+  const template =
+    (await Deno.readTextFile(join(REPO_ROOT, "template/catppuccin.user.css")))
+      .split("\n");
+  const varPfx = (variable) => `@var select ${variable}`;
+
+  for (const variable of ["darkFlavor", "lightFlavor", "accentColor"]) {
+    const expected = template.find((line) => line.includes(varPfx(variable)));
+    const current = lines.findIndex((line) => line.includes(varPfx(variable))) +
+      1;
+
+    if (current === 0) {
+      // This variable is undefined so there isn't a line for it, so we just put it at the bottom of the variables section.
+      const line = lines
+        .findLastIndex((line: string) => line.includes("==/UserStyle== */")) +
+        1;
+
+      log(
+        sprintf(
+          "Metadata variable `%s` should exist",
```

Comment:
Can we suggest what the value should be here?

Reply:
cc: @sgoudham

Reply:
Yeah it causes visual clutter since there is a GitHub user called `@var` and you're not able to escape it in the lints? and I believe there's currently no workaround.

I'm not sure if it pings them but I don't think we'd want to do it anyways, I think the current log message is _fine_. We can revisit it at a later point if the lint ends up causing too much confusion

Reply:
Alright, cool, seems fine. If you blunder the copy this bad, that's on you 👀 .